vpr/src/route/heap_type.cpp

```diff
@@ -27,6 +27,49 @@ HeapStorage::alloc() {
     temp_ptr->backward_path_cost = 0.;
     temp_ptr->R_upstream = 0.;
     temp_ptr->index = OPEN;
+    temp_ptr->path_data = nullptr;
+    temp_ptr->set_prev_node(NO_PREVIOUS);
+    temp_ptr->set_prev_edge(RREdgeId::INVALID());
+    return (temp_ptr);
+}
+
+t_heap*
+HeapStorage::alloc(bool init_data_structs) {
+    if (heap_free_head_ == nullptr) { /* No elements on the free list */
+        heap_free_head_ = vtr::chunk_new<t_heap>(&heap_ch_);
+        heap_free_head_->path_data = nullptr;
+    }
+
+    //Extract the head
+    t_heap* temp_ptr = heap_free_head_;
+    heap_free_head_ = heap_free_head_->next_heap_item();
+
+    num_heap_allocated_++;
+
+    //Reset
+    temp_ptr->set_next_heap_item(nullptr);
+    temp_ptr->cost = 0.;
+    temp_ptr->backward_path_cost = 0.;
+    temp_ptr->R_upstream = 0.;
+    temp_ptr->index = OPEN;
+
+    if (init_data_structs) {
+        // if (heap_path_free_head_ == nullptr) {
+        //     heap_path_free_head_ = vtr::chunk_new<t_heap_path>(&heap_ch_path_);
+        // }
+        if (temp_ptr->path_data == nullptr) {
+            temp_ptr->path_data = new t_heap_path;
```

Comment:
Why is the plan to use raw new and delete here rather than a alloc pool or an Arena?

Reply:
There's no reason other than it was easier and a temporary solution, and this is one of the issues I was aware of. You're definitely right that it would be faster to use said alloc pool!